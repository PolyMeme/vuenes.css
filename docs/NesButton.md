# nes-button 

Vue Component implementation of the Nes Button 

- **author** - Jesus Fernando Alvarez Franco 
- **license** - MIT 

## slots 

- `default` Use this slot for the content Inside the button 

## props 

- `primary` ***Boolean*** (*optional*) 

  Prop that define if the Button is type Primary. Will use the `is-primary` class 

- `success` ***Boolean*** (*optional*) 

  Prop that define if the Button is type Success. Will use the `is-success` class 

- `warning` ***Boolean*** (*optional*) 

  Prop that define if the Button is type Warning. Will use the `is-warning` class 

- `error` ***Boolean*** (*optional*) 

  Prop that define if the Button is type Error. Will use the `is-error` class 

- `disabled` ***Boolean*** (*optional*) 

  Prop that define if the Button is type Disabled. Will use the `is-disabled` class 

## computed properties 

- `buttonClasses` 

  Computed Property that uses the button classes depending of the props used. 

   **dependencies:** `primary`, `success`, `warning`, `error`, `disabled` 


## events 

- `click` 

  Fired when a the user clicks the button. 

## methods 

- `click()` 

  Method that emits the click event 

