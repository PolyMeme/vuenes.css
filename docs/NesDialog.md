# nes-dialog 

Vue Component implementation of the Nes Dialog 

- **author** - Jesus Fernando Alvarez Franco 
- **license** - MIT 

## slots 

- `default` Slot used for the Dialog Content 

## props 

- `rounded` ***Boolean*** (*optional*) `default: false` 

  Define if the dialog has rounded corners. Will use the `is-rounded` class 

- `dark` ***Boolean*** (*optional*) `default: false` 

  Define if the dialog is dark. Will use the `is-dark` class 

- `open` ***Boolean*** (*optional*) `default: false` 

  Define when the dialog is going to be opened. Will use the native open dialog property 

## computed properties 

- `dialogClasses` 

  Computed Property that uses the container classes depending of the props used. 

   **dependencies:** `rounded`, `dark` 

- `openCloseToggle` 

   **dependencies:** `open` 


