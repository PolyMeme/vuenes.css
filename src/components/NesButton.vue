<template>
  <button
    class="nes-btn"
    :class="buttonClasses"
    type="button"
    @click="click"
  >
    <!-- Use this slot for the content Inside the button -->
    <slot />
  </button>
</template>

<script>
/**
 * Vue Component implementation of the Nes Button
 * @author Jesus Fernando Alvarez Franco
 * @license MIT
 */
export default {
  props: {
    /**
     * Prop that define if the Button is type Primary. Will use the `is-primary` class
     * @default false
     */
    primary: Boolean,
    /**
     * Prop that define if the Button is type Success. Will use the `is-success` class
     * @default false
     */
    success: Boolean,
    /**
     * Prop that define if the Button is type Warning. Will use the `is-warning` class
     * @default false
     */
    warning: Boolean,
    /**
     * Prop that define if the Button is type Error. Will use the `is-error` class
     * @default false
     */
    error: Boolean,
    /**
     * Prop that define if the Button is type Disabled. Will use the `is-disabled` class and will not trigger the click event
     * @default false
     */
    disabled: Boolean
  },
  computed: {
    /**
     * Computed Property that uses the button classes depending of the props used.
     * @returns {{'is-primary': BooleanConstructor | *, 'is-error': BooleanConstructor | *, 'is-success': BooleanConstructor | *, 'is-disabled': BooleanConstructor | *, 'is-warning': BooleanConstructor | *}}
     */
    buttonClasses () {
      return {
        'is-primary': this.primary,
        'is-success': this.success,
        'is-warning': this.warning,
        'is-error': this.error,
        'is-disabled': this.disabled
      }
    }
  },
  methods: {
    /**
     * Method that emits the click event
     */
    click () {
      /**
       * Fired when a the user clicks the button and the button is not disabled
       */
      if (!this.disabled) {
        this.$emit('click')
      }
    }
  }
}
</script>
