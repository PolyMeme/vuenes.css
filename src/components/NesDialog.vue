<template>
  <dialog
    :class="dialogClasses"
    class="nes-dialog"
    :open="openCloseToggle"
  >
    <!-- Slot used for the Dialog Content -->
    <slot />
  </dialog>
</template>

<script>
/**
 * Vue Component implementation of the Nes Dialog
 * @author Jesus Fernando Alvarez Franco
 * @license MIT
 */
export default {
  props: {
    /**
     * Define if the dialog has rounded corners. Will use the `is-rounded` class
     * @default false
     */
    rounded: Boolean,
    /**
     * Define if the dialog is dark. Will use the `is-dark` class
     * @default false
     */
    dark: Boolean,
    /**
     * Define when the dialog is going to be opened. Will use the native open dialog property
     * @default false
     */
    open: Boolean
  },
  computed: {
    /**
     * Computed Property that uses the container classes depending of the props used.
     * @returns {{'is-rounded': BooleanConstructor, 'is-dark': BooleanConstructor}}
     */
    dialogClasses () {
      return {
        'is-rounded': this.rounded,
        'is-dark': this.dark
      }
    },
    openCloseToggle: {
      get () {
        return this.open
      },
      set (value) {
        this.$emit('update:open', value)
      }
    }
  }
}
</script>
