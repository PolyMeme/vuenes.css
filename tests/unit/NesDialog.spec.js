import { mount } from '@vue/test-utils'
import NesDialog from '../../src/components/NesDialog.vue'
import NesButton from '../../src/components/NesButton.vue'

describe('NesDialog.vue', () => {
  const createDialogWrapper = (propsData = {}, slot) => {
    return mount(NesDialog, {
      propsData,
      slots: {
        default: slot || 'This is a dialog!'
      }
    })
  }

  it('should render component properly without props', () => {
    const wrapper = createDialogWrapper()

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.find('dialog.nes-dialog').exists()).toBe(true)
    expect(wrapper.text()).toBe('This is a dialog!')
  })

  it('should set "is-rounded" class if the rounded prop is set', () => {
    const wrapper = createDialogWrapper({
      rounded: true
    })

    const roundedDialog = wrapper.find('dialog.nes-dialog.is-rounded')

    expect(roundedDialog.classes()).toContainEqual('is-rounded')
    expect(roundedDialog.exists()).toBe(true)
  })

  it('should set "is-dark" class if the dark prop is set', () => {
    const wrapper = createDialogWrapper({
      dark: true
    })

    const darkDialog = wrapper.find('dialog.nes-dialog.is-dark')

    expect(darkDialog.classes()).toContainEqual('is-dark')
    expect(darkDialog.exists()).toBe(true)
  })

  it('should set "is-dark" and "is-rounded" class if the dark and rounded prop is set', () => {
    const wrapper = createDialogWrapper({
      dark: true,
      rounded: true
    })

    const darkAndRoundedDialog = wrapper.find('dialog.nes-dialog.is-dark.is-rounded')
    const classes = darkAndRoundedDialog.classes()

    expect(classes).toContainEqual('is-dark')
    expect(classes).toContainEqual('is-rounded')
    expect(darkAndRoundedDialog.exists()).toBe(true)
  })

  it('should render slot', () => {
    const wrapper = createDialogWrapper({
      dark: true
    }, NesButton)

    expect(wrapper.find('dialog.nes-dialog.is-dark>button[type="button"].nes-btn').exists()).toBe(true)
  })

  it('should match snapshot', () => {
    const wrapper = createDialogWrapper({
      rounded: true,
      dark: true,
      open: true
    }, NesButton)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
