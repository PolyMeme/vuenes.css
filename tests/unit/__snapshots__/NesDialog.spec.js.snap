// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesDialog.vue should match snapshot 1`] = `<dialog open="open" class="nes-dialog is-rounded is-dark"><button type="button" class="nes-btn"></button></dialog>`;
